src/README.md

```diff
@@ -0,0 +1,753 @@
+# Node.js C++ codebase
+
+Hi! 👋 You’ve found the C++ code backing Node.js. This README aims to help you
+get started working on it and document some idioms you may encounter while
+doing so.
+
+## Coding style
+
+Node.js has a document detailing its [C++ coding style][]
+that can be helpful as a reference for stylistic issues.
+
+## V8 API documentation
+
+A lot of the Node.js codebase is around what the underlying JavaScript engine,
+V8, provides through its API for embedders. Knowledge of this API can also be
+useful when working with native addons for Node.js written in C++, although for
+new projects [N-API][] is typically the better alternative.
+
+V8 does not provide much public API documentation beyond what is
+available in its C++ header files, most importantly `v8.h`, which can be
+accessed online in the following locations:
+
+* On GitHub: [`v8.h` in Node.js master][]
+* On GitHub: [`v8.h` in V8 master][]
+* On the Chromium project’s Code Search application: [`v8.h` in Code Search][]
+
+V8 also provides an [introduction for V8 embedders][],
+which can be useful for understanding some of the concepts it uses in its
+embedder API.
+
+Important concepts when using V8 are the ones of [`Isolate`][]s and
+[JavaScript value handles][].
+
+## Libuv API documentation
+
+The other major dependency of Node.js is [libuv][], providing
+the [event loop][] and other operation system abstractions to Node.js.
+
+There is a [reference documentation for the libuv API][].
+
+## Helpful concepts
+
+A number of concepts are involved in putting together Node.js on top of V8 and
+libuv. This section aims to explain some of them and how they work together.
+
+<a id="isolate"></a>
+### `Isolate`
+
+The `v8::Isolate` class represents a single JavaScript engine instance, in
+particular a set of JavaScript objects that can refer to each other
+(the “heap”).
+
```

Comment:
could we also have a one liner about what APIs this class exposes?

Reply:
I’ve added a sentence but I don’t quite know what you had in mind here

Reply:
this is fine, but how about this: `This class provides important functions that control the behavior of the underlying Javascript engine (for example limit the stack size for this JS thread ), and functions that query the runtime's internal state (for example get heap statistics)`.

Reply:
@gireeshpunathil I’ve expanded the sentence with “or querying about its
current state or statistics such as memory usage”.

I don’t think we need to expand on this a ton, the most common usages of Isolate methods inside our code base are just getting the current `v8::Context` or throwing exceptions.